### PR TITLE
feat: support javascript for config lookup

### LINKED ISSUE #687: Config lookup by expression
Support arbitrary Templates for property config lookups, 

e.g.  instead of

```yaml
     - name: Version
            headline: true
            configLookup:
              config:
                type: File
                name: Configuration.properties
                labels:
                  namespace: ${.properties.namespace}
              field: $["application.buildversion"]
```

```yaml
   -  name: Version
      template:
          javascript: configs['File/Configuration.properties'].config['application.buildversion']
```

Fields that should be available

* self
* self.properties
* configs ( A map of all related configs with `type/name` key )
*

### PR DESCRIPTION
**Pull request by yashmehrotra**
Fixes #687

### CODE REVIEW

File: fixtures/topology/k8s-system.yaml [hunk 1]
```diff
@@ -36,6 +36,10 @@ spec:
                 name: nodemetrics
                 display:
                   javascript: JSON.stringify(k8s.getNodeMetrics(results))
+        - name: cluster-name
+          configLookup:
+            display:
+              javascript: config.getConfigItem("EKS/flanksource-canary-cluster").properties["name"]
```
Comment (moshloop) [inline fixtures/topology/k8s-system.yaml:42]: GetConfigItem by type/name is useful, but the primary use case would be to get the config item related to the component

File: templating/template.go [hunk 2]
```diff
@@ -42,6 +44,31 @@ func Template(environment map[string]interface{}, template v1.Template) (string,
...
+			}
+			configItem, err := db.FindConfig(configItemParams)
+			if err != nil {
+				logger.Errorf("Error fetching config item for js: %v", err)
+				emptyObj, _ := vm.ToValue(map[string]string{})
+				return emptyObj
+			}
+
+			configObject := map[string]interface{}{
+				"configs":    *configItem.Spec,
```
Comment (moshloop) [inline templating/template.go:62]: Why can't we just return the configItem  directly ?

File: fixtures/topology/k8s-system.yaml (hunk 1 above)
Comment (yashmehrotra) [reply fixtures/topology/k8s-system.yaml:42]: configItemByType("EKS") -> [array ConfigItems]
configGetItem()

File: templating/template.go (hunk 2 above)
Comment (yashmehrotra) [reply templating/template.go:62]: @moshloop In the issue description, you mentioned we should expose keys `properties` and `configs`

File: fixtures/topology/k8s-system.yaml [hunk 3]
```diff
@@ -36,6 +36,15 @@ spec:
                 name: nodemetrics
                 display:
                   javascript: JSON.stringify(k8s.getNodeMetrics(results))
+        - name: cluster-name-1
+          configLookup:
+            display:
+              javascript: configGetItem("EKS", "flanksource-canary-cluster").properties["name"]
```
Comment (yashmehrotra) [inline fixtures/topology/k8s-system.yaml:42]: getConfigItem

File: fixtures/topology/k8s-system.yaml [hunk 4]
```diff
@@ -36,6 +36,15 @@ spec:
...
+        - name: cluster-name-2
+          configLookup:
+            display:
+              javascript: configGetItemsByTypeForComponent(componentID, "EKS")[0]["name"]
```
Comment (yashmehrotra) [inline fixtures/topology/k8s-system.yaml:46]: getConfigItems("EKS")

File: pkg/topology/run.go [hunk 5]
```diff
@@ -202,6 +202,20 @@ func lookupConfig(ctx *ComponentContext, property *v1.Property, sisterProperties
 	prop := pkg.NewProperty(*property)
 
 	logger.Infof("Looking up config for %s => %s", property.Name, property.ConfigLookup.Config)
+
+	if property.ConfigLookup.Config == nil {
+		templateEnv := make(map[string]interface{})
+		if ctx.CurrentComponent != nil {
```
Comment (yashmehrotra) [inline pkg/topology/run.go:208]: don't send comp id

File: fixtures/topology/k8s-system.yaml [hunk 6]
```diff
@@ -36,6 +36,14 @@ spec:
                 name: nodemetrics
                 display:
                   javascript: JSON.stringify(k8s.getNodeMetrics(results))
+        - name: cluster-name-1
```
Comment (moshloop) [inline fixtures/topology/k8s-system.yaml:39]: Can we add a fixture for getConfigItems

**Pull request merged**